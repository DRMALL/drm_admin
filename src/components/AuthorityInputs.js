

import React from 'react'
import Select from 'antd/lib/select'
import Checkbox from 'antd/lib/checkbox'
import dispatch from '../actions/dispatch'
import store from '../commons/store'
import changeAuthIdToText from '../utils/changeAuthIdToText'
import authorityArrChange from '../utils/authorityArrChange'
import addAuthorityInput from '../utils/addAuthorityInput'
import authCheckChange from '../utils/authCheckChange'

const Option = Select.Option
export default props => {
  const { userNameArr, machineNameArr, newAuthorityArr, num, pathname } = props
  // console.log('xxx',newAuthorityArr)
  return (
    <div className='authority-input-container' >
      <div className='authority-input-flex' >
        <div className='authority-input-text' >选择设备</div>
        <Select
          showSearch
          style={{ width: 300 }}
          placeholder="选择设备"
          optionFilterProp="children"
          value={ changeAuthIdToText(newAuthorityArr[num].deviceId, machineNameArr) }
          onChange={ authorityArrChange }
          filterOption={(input, option) => option.props.children.toLowerCase().indexOf(input.toLowerCase()) >= 0}
        >
        {
          machineNameArr.map(item =>
            <Option value={ num + '-' + item._id } key={ item._id } >{ item.name }</Option>)
        }
        </Select>
      </div>
      <div className='authority-input-flex' >
        <div className='authority-input-text' >选择权限</div>
        <Checkbox onChange={ authCheckChange } id='canView' num={ num } checked={ newAuthorityArr[num].canView } >查看权限</Checkbox>
        <Checkbox onChange={ authCheckChange } id='canMonitor' num={ num } checked={ newAuthorityArr[num].canMonitor } >监控权限</Checkbox>
      </div>
      <div className='authority-input-flex' >
        <div className='authority-input-text' >选择用户</div>
        <Select
          showSearch
          // mode='multiple'
          style={{ width: 300 }}
          onDeselect={ delSelectUser }
          onSelect={ addSelectUser }
          placeholder="选择用户"
          optionFilterProp="children"
          value={ changeAuthIdToText(newAuthorityArr[num].userId, userNameArr ) }
          // onChange={handleChange}
          filterOption={(input, option) => option.props.children.toLowerCase().indexOf(input.toLowerCase()) >= 0}
        >
        {
          userNameArr.map(item =>
            <Option value={ num + '-' + item._id } key={ item._id } >{ item.name }</Option>)
        }
        </Select>
        <div style={{ visibility: (newAuthorityArr.length -1 === num && pathname !=='/authority/edit') ? 'visible' : 'hidden'}} className='authority-input-icon'
             onClick={ addAuthorityInput } >
          <i className="iconfont icon-tianjiajiahaowubiankuang" style={{ width:12,height:12, color: '#fff' }}  ></i>
        </div>
      </div>
    </div>
    )
}


function addSelectUser(e){
  //目前为单选
  const { newAuthorityArr } = store.getState().authority
    let arr = e.split('-')
    let num = parseInt(arr[0], 10)
    newAuthorityArr[num].userId = arr[1]
    dispatch('AUTHORITY_GET_MACHINE_VALUE', newAuthorityArr )
}

function delSelectUser(e){
  //目前为单选,此函数无用
  const { newAuthorityArr } = store.getState().authority
    let arr = e.split('-')
    let num = parseInt(arr[0], 10)
    newAuthorityArr[num].userId = ''
    dispatch('AUTHORITY_GET_MACHINE_VALUE', newAuthorityArr )
}

