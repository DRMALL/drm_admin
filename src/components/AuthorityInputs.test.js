

import AuthorityInputs from './AuthorityInputs'
import React from 'react'
import renderer from 'react-test-renderer'



it('AuthorityInputs test', () => {
  const newAuthorityArr = [
  {canMonitor:false, canView:false, deviceId:"",userId: ''}
  ]
  const machineNameArr = [
  {name: '设备名称', _id: '34534e35' },
  {name: '钻井', _id: '34534e36' },
  {name: '石油开采', _id: '34534e37' }
  ]
  const userNameArr = [
  {name: '张三', _id: '3453fe34' },
  {name: '李四', _id: '3453fe35' },
  ]
  const pathname = '/authority/new'
  const num = 0
  const index = 0
  const tree =  renderer.create(<AuthorityInputs pathname={ pathname }
                                newAuthorityArr={ newAuthorityArr }
                                userNameArr={ userNameArr }
                                machineNameArr={ machineNameArr }
                                num={ index }
                                key={index}/>)
  expect(tree).toMatchSnapshot()
})
