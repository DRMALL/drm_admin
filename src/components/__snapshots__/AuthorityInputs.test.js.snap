// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorityInputs test 1`] = `
<div
  className="authority-input-container"
>
  <div
    className="authority-input-flex"
  >
    <div
      className="authority-input-text"
    >
      选择设备
    </div>
    <div
      className="ant-select ant-select-enabled"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={undefined}
      onMouseLeave={undefined}
      onTouchStart={[Function]}
      style={
        Object {
          "width": 300,
        }
      }
    >
      <div
        aria-autocomplete="list"
        aria-expanded={false}
        aria-haspopup="true"
        className="ant-select-selection
                  ant-select-selection--single"
        onKeyDown={[Function]}
        role="combobox"
        tabIndex={0}
      >
        <div
          className="ant-select-selection__rendered"
        >
          <div
            className="ant-select-selection__placeholder"
            onClick={[Function]}
            onMouseDown={[Function]}
            style={
              Object {
                "WebkitUserSelect": "none",
                "display": "none",
                "userSelect": "none",
              }
            }
            unselectable="unselectable"
          >
            选择设备
          </div>
          <div
            className="ant-select-selection-selected-value"
            style={
              Object {
                "display": "block",
                "opacity": 1,
              }
            }
            title={null}
          />
          <div
            className="ant-select-search ant-select-search--inline"
            style={
              Object {
                "display": "none",
              }
            }
          >
            <div
              className="ant-select-search__field__wrap"
            >
              <input
                className="ant-select-search__field"
                disabled={undefined}
                onChange={[Function]}
                onKeyDown={[Function]}
                value=""
              />
              <span
                className="ant-select-search__field__mirror"
              >
                
                 
              </span>
            </div>
          </div>
        </div>
        <span
          className="ant-select-arrow"
          onClick={[Function]}
          style={
            Object {
              "WebkitUserSelect": "none",
              "userSelect": "none",
            }
          }
          unselectable="unselectable"
        >
          <b />
        </span>
      </div>
    </div>
  </div>
  <div
    className="authority-input-flex"
  >
    <div
      className="authority-input-text"
    >
      选择权限
    </div>
    <label
      className="ant-checkbox-wrapper"
      onMouseEnter={undefined}
      onMouseLeave={undefined}
      style={undefined}
    >
      <span
        className="ant-checkbox"
        style={Object {}}
      >
        <input
          checked={false}
          className="ant-checkbox-input"
          disabled={undefined}
          name={undefined}
          onBlur={[Function]}
          onChange={[Function]}
          onClick={undefined}
          onFocus={[Function]}
          readOnly={undefined}
          tabIndex={undefined}
          type="checkbox"
        />
        <span
          className="ant-checkbox-inner"
        />
      </span>
      <span>
        查看权限
      </span>
    </label>
    <label
      className="ant-checkbox-wrapper"
      onMouseEnter={undefined}
      onMouseLeave={undefined}
      style={undefined}
    >
      <span
        className="ant-checkbox"
        style={Object {}}
      >
        <input
          checked={false}
          className="ant-checkbox-input"
          disabled={undefined}
          name={undefined}
          onBlur={[Function]}
          onChange={[Function]}
          onClick={undefined}
          onFocus={[Function]}
          readOnly={undefined}
          tabIndex={undefined}
          type="checkbox"
        />
        <span
          className="ant-checkbox-inner"
        />
      </span>
      <span>
        监控权限
      </span>
    </label>
  </div>
  <div
    className="authority-input-flex"
  >
    <div
      className="authority-input-text"
    >
      选择用户
    </div>
    <div
      className="ant-select ant-select-enabled"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={undefined}
      onMouseLeave={undefined}
      onTouchStart={[Function]}
      style={
        Object {
          "width": 300,
        }
      }
    >
      <div
        aria-autocomplete="list"
        aria-expanded={false}
        aria-haspopup="true"
        className="ant-select-selection
                  ant-select-selection--single"
        onKeyDown={[Function]}
        role="combobox"
        tabIndex={0}
      >
        <div
          className="ant-select-selection__rendered"
        >
          <div
            className="ant-select-selection__placeholder"
            onClick={[Function]}
            onMouseDown={[Function]}
            style={
              Object {
                "WebkitUserSelect": "none",
                "display": "none",
                "userSelect": "none",
              }
            }
            unselectable="unselectable"
          >
            选择用户
          </div>
          <div
            className="ant-select-selection-selected-value"
            style={
              Object {
                "display": "block",
                "opacity": 1,
              }
            }
            title={null}
          />
          <div
            className="ant-select-search ant-select-search--inline"
            style={
              Object {
                "display": "none",
              }
            }
          >
            <div
              className="ant-select-search__field__wrap"
            >
              <input
                className="ant-select-search__field"
                disabled={undefined}
                onChange={[Function]}
                onKeyDown={[Function]}
                value=""
              />
              <span
                className="ant-select-search__field__mirror"
              >
                
                 
              </span>
            </div>
          </div>
        </div>
        <span
          className="ant-select-arrow"
          onClick={[Function]}
          style={
            Object {
              "WebkitUserSelect": "none",
              "userSelect": "none",
            }
          }
          unselectable="unselectable"
        >
          <b />
        </span>
      </div>
    </div>
    <div
      className="authority-input-icon"
      onClick={[Function]}
      style={
        Object {
          "visibility": "visible",
        }
      }
    >
      <i
        className="iconfont icon-tianjiajiahaowubiankuang"
        style={
          Object {
            "color": "#fff",
            "height": 12,
            "width": 12,
          }
        }
      />
    </div>
  </div>
</div>
`;
